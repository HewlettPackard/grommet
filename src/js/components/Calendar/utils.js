
// Utility functions for the Calendar.
// Just what's needed to avoid having to include a dependency like momentjs.

const DAY_MILLISECONDS = 24 * 60 * 60 * 1000;

export const addDays = (date, days) =>
  (new Date(date.getTime() + (DAY_MILLISECONDS * days)));

export const subtractDays = (date, days) => addDays(date, -days);

export const addMonths = (date, months) => {
  const result = new Date(date);
  const years = Math.floor((date.getMonth() + months) / 12);
  result.setFullYear(date.getFullYear() + years);
  result.setMonth((date.getMonth() + months) % 12);
  return result;
};

export const subtractMonths = (date, months) => addMonths(date, -months);

export const sameDay = (date1, date2) => (
  date1.getFullYear() === date2.getFullYear() &&
  date1.getMonth() === date2.getMonth() &&
  date1.getDate() === date2.getDate()
);

export const sameDayOrAfter = (date1, date2) =>
  (date1.getFullYear() > date2.getFullYear() ||
    (date1.getFullYear() === date2.getFullYear() && (
      date1.getMonth() > date2.getMonth() || (
        date1.getMonth() === date2.getMonth() && (date1.getDate() >= date2.getDate())
      )
    ))
  );

export const sameDayOrBefore = (date1, date2) =>
  (date1.getFullYear() < date2.getFullYear() ||
    (date1.getFullYear() === date2.getFullYear() && (
      date1.getMonth() < date2.getMonth() || (
        date1.getMonth() === date2.getMonth() && (date1.getDate() <= date2.getDate())
      )
    ))
  );

export const daysApart = (date1, date2) =>
  Math.floor((date1.getTime() - date2.getTime()) / DAY_MILLISECONDS);

// betweenDates takes and array of two elements and checks if the
// supplied date lies between them, inclusive.
// returns 2 if exact match to one end, 1 if between, undefined otherwise
export const betweenDates = (date, dates) => {
  let result;
  if (dates) {
    const [from, to] = dates.map(d => new Date(d));
    if (sameDay(date, from) || sameDay(date, to)) {
      result = 2;
    } else if (sameDayOrAfter(date, from) && sameDayOrBefore(date, to)) {
      result = 1;
    }
  } else {
    result = 1;
  }
  return result;
};

// withinDates takes and array of string dates or 2 element arrays and
// checks whether the supplied date matches any string or is between
// any dates in arrays.
// returns 2 if exact match, 1 if between, undefined otherwise
export const withinDates = (date, dates) => {
  let result;
  if (dates) {
    if (Array.isArray(dates)) {
      dates.some((d) => {
        if (typeof d === 'string') {
          if (sameDay(date, new Date(d))) {
            result = 2;
          }
        } else {
          result = betweenDates(date, d);
        }
        return result;
      });
    } else if (sameDay(date, new Date(dates))) {
      result = 2;
    }
  }
  return result;
};
