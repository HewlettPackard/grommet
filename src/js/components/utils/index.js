export * from './styles';
export * from './mixins';
export * from './DOM';
export { default as Drop } from './Drop';
