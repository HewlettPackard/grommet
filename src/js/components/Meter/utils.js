import { colorForName } from '../../utils';

export const strokeProps = (color, theme) => {
  const result = {};
  if (color) {
    if (typeof color === 'object') {
      result.stroke = colorForName(color.color, theme);
      if (color.opacity) {
        result.strokeOpacity = (color.opacity === true ?
          theme.global.opacity.medium : theme.global.opacity[color.opacity]);
      }
    } else {
      result.stroke = colorForName(color, theme);
    }
  }
  return result;
};

export default { strokeProps };
