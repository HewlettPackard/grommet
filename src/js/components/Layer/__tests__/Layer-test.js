import React, { Component } from 'react';
import 'jest-styled-components';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { createPortal, expectPortal } from '../../../utils/portal';

import { Grommet } from '../../Grommet';
import { Layer, LayerContainer } from '../';

Enzyme.configure({ adapter: new Adapter() });

class FakeLayer extends Component {
  state = {
    showLayer: false,
  }
  componentDidMount() {
    /* eslint-disable react/no-did-mount-set-state */
    this.setState({ showLayer: true });
    /* eslint-enable react/no-did-mount-set-state */
  }
  render() {
    const { children, hide } = this.props;
    const { showLayer } = this.state;
    let layer;
    if (!hide && showLayer) {
      layer = (
        <Layer onEsc={() => this.setState({ showLayer: false })}>
          <div id='layer-node'>
            This is a layer
            <input />
          </div>
        </Layer>
      );
    }
    return (
      <Grommet>
        {layer}
        {children}
      </Grommet>
    );
  }
}

describe('Layer', () => {
  beforeEach(createPortal);

  ['top', 'bottom', 'left', 'right', 'center'].forEach(position => (
    [true, false, 'horizontal', 'vertical'].forEach(full => (
      ['none', 'xsmall', 'small', 'medium', 'large'].forEach(margin => (
        test(`position ${position} full ${full} margin ${margin}`, () => {
          mount(
            <Layer id='position-test' position={position} full={full} margin={margin}>
              This is a layer
            </Layer>
          );

          expectPortal('position-test').toMatchSnapshot();
        })
      ))
    ))
  ));

  test('hides', () => {
    const component = mount(
      <Layer id='hidden-test' position='hidden'>
        This is a layer
      </Layer>
    );

    expectPortal('hidden-test').toMatchSnapshot();

    component.setProps({ hidden: false });

    expectPortal('hidden-test').toMatchSnapshot();
  });

  test('plain renders', () => {
    mount(
      <Layer id='plain-test' plain={true}>
        This is a plain layer
      </Layer>
    );

    expectPortal('plain-test').toMatchSnapshot();
  });

  test('non-modal renders', () => {
    mount(
      <Layer id='non-modal-test' modal={false}>
        This is a non-modal layer
      </Layer>
    );

    expectPortal('non-modal-test').toMatchSnapshot();
  });

  test('invokes onEsc', () => {
    const onEsc = jest.fn();
    const component = mount(
      <Grommet>
        <LayerContainer onEsc={onEsc}>
          <input />
        </LayerContainer>
      </Grommet>
    );
    component.find('input').simulate('keyDown', { key: 'Esc', keyCode: 27, which: 27 });
    expect(onEsc).toBeCalled();
    component.unmount();
  });

  test('is accessible', () => {
    // make sure to remove all body children
    document.body.innerHTML = '';
    document.body.appendChild(document.createElement('div'));
    /* eslint-disable jsx-a11y/tabindex-no-positive */
    const component = mount(
      <FakeLayer>
        <div id='body-node'>
          <input />
          <input tabIndex='10' />
        </div>
      </FakeLayer>, {
        attachTo: document.body.firstChild,
      }
    );
    /* eslint-enable jsx-a11y/tabindex-no-positive */

    let bodyNode = component.find('#body-node').getDOMNode();
    const layerNode = document.getElementById('layer-node');
    expect(bodyNode).toMatchSnapshot();
    expect(layerNode).toMatchSnapshot();

    component.setProps({ hide: true });

    bodyNode = component.find('#body-node').getDOMNode();
    expect(bodyNode).toMatchSnapshot();
    expect(document.getElementById('layer-node')).toBeNull();
  });
});
