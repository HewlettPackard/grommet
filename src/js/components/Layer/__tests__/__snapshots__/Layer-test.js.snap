// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer aligns bottom 1`] = `
<div
  class="StyledLayer-EylWz jhJjNf"
  id="bottom-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv ilWCsF"
  >
    This is a layer
  </div>
</div>
`;

exports[`Layer aligns bottom 2`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .jvBdNm {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .jvBdNm {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    -webkit-animation: slide-right 0.2s ease-in-out forwards;
    animation: slide-right 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}

@keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .hUdiMG {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .hUdiMG {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
    -webkit-animation: slide-down 0.2s ease-in-out forwards;
    animation: slide-down 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}

@keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .ilWCsF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .ilWCsF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    bottom: 0px;
    right: 50%;
    -webkit-transform: translateX(50%);
    -ms-transform: translateX(50%);
    transform: translateX(50%);
    -webkit-animation: slide-up 0.2s ease-in-out forwards;
    animation: slide-up 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}

@keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}
}



.jhJjNf {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0,0,0,0.5);
}

@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}"
`;

exports[`Layer aligns left 1`] = `
<div
  class="StyledLayer-EylWz jhJjNf"
  id="left-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv cuhhFF"
  >
    This is a layer
  </div>
</div>
`;

exports[`Layer aligns left 2`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}



.jhJjNf {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0,0,0,0.5);
}

@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}"
`;

exports[`Layer aligns right 1`] = `
<div
  class="StyledLayer-EylWz jhJjNf"
  id="right-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv jvBdNm"
  >
    This is a layer
  </div>
</div>
`;

exports[`Layer aligns right 2`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .jvBdNm {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .jvBdNm {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    -webkit-animation: slide-right 0.2s ease-in-out forwards;
    animation: slide-right 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}

@keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}
}



.jhJjNf {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0,0,0,0.5);
}

@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}"
`;

exports[`Layer aligns top 1`] = `
<div
  class="StyledLayer-EylWz jhJjNf"
  id="top-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv hUdiMG"
  >
    This is a layer
  </div>
</div>
`;

exports[`Layer aligns top 2`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .jvBdNm {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .jvBdNm {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    -webkit-animation: slide-right 0.2s ease-in-out forwards;
    animation: slide-right 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}

@keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .hUdiMG {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .hUdiMG {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
    -webkit-animation: slide-down 0.2s ease-in-out forwards;
    animation: slide-down 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}

@keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}
}



.jhJjNf {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0,0,0,0.5);
}

@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}"
`;

exports[`Layer hides 1`] = `
<div
  class="StyledLayer-EylWz hAhzzE"
  id="hidden-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv duLPUq"
  >
    This is a layer
  </div>
</div>
`;

exports[`Layer hides 2`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .jvBdNm {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .jvBdNm {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    -webkit-animation: slide-right 0.2s ease-in-out forwards;
    animation: slide-right 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}

@keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .hUdiMG {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .hUdiMG {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
    -webkit-animation: slide-down 0.2s ease-in-out forwards;
    animation: slide-down 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}

@keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .ilWCsF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .ilWCsF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    bottom: 0px;
    right: 50%;
    -webkit-transform: translateX(50%);
    -ms-transform: translateX(50%);
    transform: translateX(50%);
    -webkit-animation: slide-up 0.2s ease-in-out forwards;
    animation: slide-up 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}

@keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .duLPUq {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .duLPUq {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
  }
}



@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}

.hAhzzE {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0,0,0,0.5);
  left: -100%;
  right: 100%;
  z-index: -1;
  position: fixed;
}"
`;

exports[`Layer hides 3`] = `
<div
  class="StyledLayer-EylWz hAhzzE"
  id="hidden-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv duLPUq"
  >
    This is a layer
  </div>
</div>
`;

exports[`Layer hides 4`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .jvBdNm {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .jvBdNm {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    -webkit-animation: slide-right 0.2s ease-in-out forwards;
    animation: slide-right 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}

@keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .hUdiMG {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .hUdiMG {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
    -webkit-animation: slide-down 0.2s ease-in-out forwards;
    animation: slide-down 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}

@keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .ilWCsF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .ilWCsF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    bottom: 0px;
    right: 50%;
    -webkit-transform: translateX(50%);
    -ms-transform: translateX(50%);
    transform: translateX(50%);
    -webkit-animation: slide-up 0.2s ease-in-out forwards;
    animation: slide-up 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}

@keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .duLPUq {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .duLPUq {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
  }
}



@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}

.hAhzzE {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: rgba(0,0,0,0.5);
  left: -100%;
  right: 100%;
  z-index: -1;
  position: fixed;
}"
`;

exports[`Layer is accessible 1`] = `
<div
  id="body-node"
>
  <input
    tabindex="-1"
  />
  <input
    tabindex="-1"
    data-g-tabindex="10"
  />
</div>
`;

exports[`Layer is accessible 2`] = `
<div
  id="layer-node"
>
  This is a layer
  <input />
</div>
`;

exports[`Layer is accessible 3`] = `
<div
  id="body-node"
>
  <input />
  <input
    tabindex="10"
  />
</div>
`;

exports[`Layer plain renders 1`] = `
<div
  class="StyledLayer-EylWz juFlxl"
  id="plain-test"
  tabindex="-1"
>
  <div
    class="StyledLayer__StyledContainer-iKmEpv eNADdO"
  >
    This is a plain layer
  </div>
</div>
`;

exports[`Layer plain renders 2`] = `
"@media only screen and (max-width:699px) {
  .cuhhFF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .cuhhFF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    left: 0px;
    -webkit-animation: slide-left 0.2s ease-in-out forwards;
    animation: slide-left 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}

@keyframes slide-left {
    0% {
      left: -100%;
    }

    100% {
      left: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .jvBdNm {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .jvBdNm {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    top: 0px;
    bottom: 0px;
    right: 0px;
    -webkit-animation: slide-right 0.2s ease-in-out forwards;
    animation: slide-right 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}

@keyframes slide-right {
    0% {
      right: -200px;
    }

    100% {
      right: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .hUdiMG {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .hUdiMG {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    left: 50%;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
    -webkit-animation: slide-down 0.2s ease-in-out forwards;
    animation: slide-down 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}

@keyframes slide-down {
    0% {
      top: -100vh;
    }

    100% {
      top: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .ilWCsF {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .ilWCsF {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
    bottom: 0px;
    right: 50%;
    -webkit-transform: translateX(50%);
    -ms-transform: translateX(50%);
    transform: translateX(50%);
    -webkit-animation: slide-up 0.2s ease-in-out forwards;
    animation: slide-up 0.2s ease-in-out forwards;
  }

@-webkit-keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}

@keyframes slide-up {
    0% {
      margin-bottom: -200px;
    }

    100% {
      margin-bottom: 0px;
    }
}
}

@media only screen and (max-width:699px) {
  .duLPUq {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .duLPUq {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: 4px;
  }
}

@media only screen and (max-width:699px) {
  .eNADdO {
    min-height: 100%;
    min-width: 100%;
  }
}

@media only screen and (min-width:700px) {
  .eNADdO {
    position: absolute;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    border-radius: none;
    bottom: 50%;
    right: 50%;
    -webkit-animation: grow-box 0.1s forwards;
    animation: grow-box 0.1s forwards;
  }

@-webkit-keyframes grow-box {
    0% {
      -webkit-transform: translate(50%,50%) scale(0.8);
      -ms-transform: translate(50%,50%) scale(0.8);
      transform: translate(50%,50%) scale(0.8);
    }

    100% {
      -webkit-transform: translate(50%,50%) scale(1);
      -ms-transform: translate(50%,50%) scale(1);
      transform: translate(50%,50%) scale(1);
    }
}

@keyframes grow-box {
    0% {
      -webkit-transform: translate(50%,50%) scale(0.8);
      -ms-transform: translate(50%,50%) scale(0.8);
      transform: translate(50%,50%) scale(0.8);
    }

    100% {
      -webkit-transform: translate(50%,50%) scale(1);
      -ms-transform: translate(50%,50%) scale(1);
      transform: translate(50%,50%) scale(1);
    }
}
}



@media only screen and (min-width:700px) {
  .jhJjNf {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}

.juFlxl {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  position: relative;
  z-index: 10;
  height: 100vh;
  overflow: auto;
  background-color: transparent;
}

@media only screen and (min-width:700px) {
  .juFlxl {
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
}"
`;
